src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs

```diff
@@ -1675,6 +1675,70 @@ public SyntaxNode TypeExpression(ITypeSymbol typeSymbol, bool addImport)
         public abstract SyntaxNode NullableTypeExpression(SyntaxNode type);
 
         /// <summary>
+        /// Creates an expression that denotes a tuple type.
+        /// </summary>
+        public abstract SyntaxNode TupleTypeExpression(IEnumerable<SyntaxNode> elements);
+
+        protected static void ValidateTupleElements(IEnumerable<SyntaxNode> elements)
```

Comment:
A simple way to do this is to have TupleTypeExpression be non abstract.  It will check the elements, and if they're ok, it will call an abstract method to actually make the syntax.  the Vb and C# versions just make the syntax and don't have to do any checking.